app/cdap/components/ConnectionList/index.tsx

```diff
@@ -145,42 +158,111 @@ export default function ConnectionList() {
     });
   };
 
+  const makeCursorFocused = (index: number) => {
+    refs.current[index].focus();
+  };
+
   const selectedTabValueHandler = (entity: IRecords, index: number) => {
     toggleLoader(true);
     setDataForTabs((currentData) => {
       let newData = [...currentData];
       newData[index].selectedTab = entity.name;
-      newData = newData.map((each) => {
+      newData = newData.map((eachNewData) => {
         return {
-          data: each.data,
-          showTabs: each.showTabs,
-          selectedTab: each.selectedTab,
+          data: eachNewData.data,
+          showTabs: eachNewData.showTabs,
+          selectedTab: eachNewData.selectedTab,
           isSearching: false,
         };
       });
       if (index === 0) {
         getCategorizedConnectionsforSelectedTab(entity.name as string, index);
       } else if (index === 1) {
-        fetchEntities(entity.name).then((res) => {
-          setDataForTabsHelper(res, index);
-          toggleLoader(false);
-        });
-      } else {
-        if (entity.canBrowse) {
-          fetchEntities(dataForTabs[1].selectedTab, entity.path as string).then((res) => {
+        fetchEntities(entity.name)
+          .then((res) => {
             setDataForTabsHelper(res, index);
             toggleLoader(false);
+          })
+          .catch((error) => {
+            toggleLoader(false);
+            setIsErrorOnNoWorkSpace(true);
+            // TODO : Need to bind the message on Snackbar . The message is in error.message same as the old UI . Remove the console log later
           });
+      } else {
+        if (entity.canBrowse) {
+          fetchEntities(dataForTabs[1].selectedTab, entity.path as string)
+            .then((res) => {
+              setDataForTabsHelper(res, index);
+              toggleLoader(false);
+            })
+            .catch((error) => {
+              toggleLoader(false);
+              setIsErrorOnNoWorkSpace(true);
+              // TODO : Need to bind the message on Snackbar. Remove the console log later
+            });
         }
       }
       return newData;
     });
   };
 
+  const searchHandler = (index: number) => {
+    setDataForTabs((prev) => {
+      let tempData = [...prev];
+      tempData = tempData.map((eachTempData) => {
+        return {
+          ...eachTempData,
+          isSearching: false,
```

Comment:
can we move logics that are not manipulating the states directly to separate utils and reuse them plz